packages/@lwc/rollup-plugin/src/index.js

```diff
@@ -31,29 +31,50 @@ function isMixingJsAndTs(importerExt, importeeExt) {
     );
 }
 
+// From a list of modules resolved, create an Map for faster moduleId seach
+function createMapFromCollectedModules(modules) {
+    return modules.reduce((map, moduleRecord) => {
+        const id = moduleRecord.specifier;
+        const entry = map.get(id) || [];
+        entry.push(moduleRecord);
+        // We need to sort them by the longest scope so we find the most specific first
```

Comment:
Can you expand on this comment? A path length is not related to the module "specificity", at best it's the number of `fs.seperator` in the path.

Reply:
I will think a bit more on this,  but I believe is sufficient because we care about the nestedness, hence length will always be correct, siblings its irrelevant because the root part of the path will be different